### ISSUE TITLE
Gradient clipping

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/litgpt#1222]
LLMs are commonly pretrained using gradient clipping, which we also implement for pretraining here in LitGPT. In addition, it might be useful to add a gradient clipping option to the finetuning scripts as well. We can set it to `None` by default so it doesn't change the current performance. However, I think it would be a nice option to have for the config files etc.
